compiler/qsc_qasm/src/semantic/lowerer.rs

```diff
@@ -206,11 +206,33 @@ impl Lowerer {
             syntax::StmtKind::ExternDecl(extern_decl) => self.lower_extern(extern_decl),
             syntax::StmtKind::For(stmt) => self.lower_for_stmt(stmt),
             syntax::StmtKind::If(stmt) => self.lower_if_stmt(stmt),
-            syntax::StmtKind::GateCall(stmt) => self.lower_gate_call_stmt(stmt),
-            syntax::StmtKind::GPhase(stmt) => self.lower_gphase_stmt(stmt),
+            syntax::StmtKind::GateCall(stmt) => {
+                let mut stmts = Vec::new();
+                for kind in self.lower_gate_call_stmt(stmt) {
+                    let stmt = semantic::Stmt {
+                        span: stmt.span,
+                        annotations: annotations.clone(),
+                        kind: Box::new(kind),
+                    };
+                    stmts.push(stmt);
+                }
+                return stmts;
+            }
+            syntax::StmtKind::GPhase(stmt) => {
+                let mut stmts = Vec::new();
+                for kind in self.lower_gphase_stmt(stmt) {
+                    let stmt = semantic::Stmt {
+                        span: stmt.span,
+                        annotations: annotations.clone(),
+                        kind: Box::new(kind),
+                    };
+                    stmts.push(stmt);
+                }
+                return stmts;
+            }
```

Comment:
Create `lower_gate_call_stmts` and `lower_gphase_stmts` instead of blocks in the mieedle of match